File: tests/fixtures/simple-node/src/echo.js
/**
 * Echo Lambda handler - returns whatever was sent in the body
 */
exports.handler = async (event, context) => {
  console.log('Echo received:', event.body);

  let body;
  try {
    body = event.body ? JSON.parse(event.body) : {};
  } catch (e) {
    body = { raw: event.body };
  }

  return {
    statusCode: 200,
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      echo: body,
      method: event.httpMethod,
      path: event.path,
      timestamp: new Date().toISOString(),
    }),
  };
};

File: tests/fixtures/simple-python/src/echo.py
"""Echo handler - returns the request body"""

import json


def handler(event, context):
    body = event.get("body")
    if body and isinstance(body, str):
        try:
            body = json.loads(body)
        except json.JSONDecodeError:
            pass

    return {
        "statusCode": 200,
        "headers": {
            "Content-Type": "application/json",
        },
        "body": json.dumps({
            "echo": body,
            "method": event.get("httpMethod", "UNKNOWN"),
        }),
    }
